sns_ik_examples/src/ik_tests.cpp

```diff
@@ -307,53 +386,82 @@ void test(ros::NodeHandle& nh, double num_samples_pos, double num_samples_vel,
+  velocitySolverData sns_fast = {sns_ik::SNS_Fast,"SNS Fast",0.0,0.0,0.0,0.0,0.0,0.0};
   vel_solver_data.push_back(sns_fast);
-  velocitySolverData sns_fastoptimal = {sns_ik::SNS_FastOptimal,"SNS Fast Optimal",0.0,0.0,0.0};
+  velocitySolverData sns_fastoptimal = {sns_ik::SNS_FastOptimal,"SNS Fast Optimal",0.0,0.0,0.0,0.0,0.0,0.0};
   vel_solver_data.push_back(sns_fastoptimal);
 
   // These values are not used yet
   double posIK_linearMaxStepSize = 0.05;
   double posIK_angularMaxStepSize = 0.05;
   double posIK_maxIterations = 150;
   double posIK_dt=0.2;
-
   for(auto& vst: vel_solver_data){
     snsik_solver.setVelocitySolveType(vst.type);
     // Beginnings of parameter setting. Right now this sets the pos solver to its defaults
     std::shared_ptr<sns_ik::SNSPositionIK> pos_solver;
     snsik_solver.getPositionSolver(pos_solver);
-    // pos solver param setting end
+    // Initialize Solver Variables
     total_time=0;
+    ns_total_time=0;
     success=0;
+    ns_success=0;
+    both_success_cnt=0;
+    total_ns_l2_norm_ratio = 0.0;
     ROS_INFO_STREAM("*** Testing SNS-IK with "<<num_samples_pos<<" random samples");
     for (uint i=0; i < num_samples_pos; i++) {
+      // Initialize Iteration Variables
+      elapsed = 0;
+      both_success = false;
+      //Solve forward kinematics for both
       fk_solver.JntToCart(JointList[i],end_effector_pose);
-      double elapsed = 0;
-
       start_time = boost::posix_time::microsec_clock::local_time();
-
-      if(nullspace_bias_task)
-        rc = snsik_solver.CartToJnt(JointSeed[i], end_effector_pose, ns_bias, result);
-      else
-        rc = snsik_solver.CartToJnt(JointSeed[i], end_effector_pose, result);
-
+      rc = snsik_solver.CartToJnt(JointSeed[i], end_effector_pose, result);
       diff = boost::posix_time::microsec_clock::local_time() - start_time;
       elapsed = diff.total_nanoseconds() / 1e9;
       total_time+=elapsed;
       vst.indiv_time.push_back(elapsed);
-      if (rc>=0)
+      fk_solver.JntToCart(result, end_effector_pose_check);
+      if (rc>=0 && in_pos_bounds(result, ll, ul)
+                && Equal(end_effector_pose, end_effector_pose_check, 1e-3)){
         success++;
+        both_success=true;
+       }
+
+      if(use_nullspace_bias_task) {
+        elapsed = 0;
+        start_time = boost::posix_time::microsec_clock::local_time();
+        ns_rc = snsik_solver.CartToJnt(JointSeed[i], end_effector_pose, NullSpaceBias[i], ns_result);
+        diff = boost::posix_time::microsec_clock::local_time() - start_time;
+        elapsed = diff.total_nanoseconds() / 1e9;
+        ns_total_time += elapsed;
+        vst.indiv_ns_time.push_back(elapsed);
+        fk_solver.JntToCart(ns_result, end_effector_pose_check);
+        if (ns_rc>=0 && in_pos_bounds(ns_result, ll, ul)
+                  && Equal(end_effector_pose, end_effector_pose_check, 1e-3)){
+          ns_success++;
+          both_success &= both_success;
```

Comment:
Same as above. This statement will have no affect and does not turn false if the NS task fails.

The easier solution would probably be to start with both_success to true and set to false if either solver failed.